Description: Using LDFLAGS and CPPFLAGSpassed down from
  dpkg-buildflags such that hardening gets enabled.
Forwarded: doesn't make sense upstream
Author: Ruben Undheim <ruben.undheim@gmail.com>
Index: berkeley-abc-1.01-20140822hg4d547a5e065b/Makefile
===================================================================
--- berkeley-abc-1.01-20140822hg4d547a5e065b.orig/Makefile	2014-09-14 12:16:38.850652511 +0200
+++ berkeley-abc-1.01-20140822hg4d547a5e065b/Makefile	2014-09-14 12:19:23.174654274 +0200
@@ -103,19 +103,19 @@
 
 %.o: %.c
 	@echo "$(MSG_PREFIX)\`\` Compiling:" $(LOCAL_PATH)/$<
-	@$(CC) -c $(CFLAGS) $< -o $@
+	@$(CC) -c $(CFLAGS) $(CPPFLAGS) $< -o $@
 
 %.o: %.cc
 	@echo "$(MSG_PREFIX)\`\` Compiling:" $(LOCAL_PATH)/$<
-	@$(CXX) -c $(CXXFLAGS) $< -o $@
+	@$(CXX) -c $(CXXFLAGS) $(CPPFLAGS) $< -o $@
 
 %.d: %.c
 	@echo "$(MSG_PREFIX)\`\` Dependency:" $(LOCAL_PATH)/$<
-	@./depends.sh $(CC) `dirname $*.c` $(CFLAGS) $*.c > $@
+	@./depends.sh $(CC) `dirname $*.c` $(CFLAGS) $(CPPFLAGS) $*.c > $@
 
 %.d: %.cc
 	@echo "$(MSG_PREFIX)\`\` Generating dependency:" $(LOCAL_PATH)/$<
-	@./depends.sh $(CXX) `dirname $*.cc` $(CXXFLAGS) $*.cc > $@
+	@./depends.sh $(CXX) `dirname $*.cc` $(CXXFLAGS) $(CPPFLAGS) $*.cc > $@
 
 -include $(DEP)
 
@@ -132,7 +132,7 @@
 
 $(PROG): $(OBJ)
 	@echo "$(MSG_PREFIX)\`\` Building binary:" $(notdir $@)
-	@$(LD) -o $@ $^ $(LIBS)
+	@$(LD) -o $@ $^ $(LIBS) $(LDFLAGS)
 
 lib$(PROG).a: $(OBJ)
 	@echo "$(MSG_PREFIX)\`\` Linking:" $(notdir $@)
